### DISCUSSION TITLE
Is it possible to force the timestamp when creating a document?

### DISCUSSION THREAD

**Discussion by wsw70** [zinclabs/zinc#229] [Q&A] [answered]
I need to import documents into my instance of `zinc` but I would like to keep the existing timeline. I use calls such as 

    PUT https://zinc.example.com/api/hello/_doc/be58b66e-9fe0-4a67-b27f-6669bddda8f0

How can I set `@timestamp` in such a call?

**Comment by wsw70** [comment, marked as answer]
I had a look at the code and, behold, this is possible :) https://github.com/zinclabs/zinc/blob/7d73d941274b00554482a10d55a6d2ee4312816a/pkg/core/index_document.go#L106

One simply need to add a `@timestamp` field with the timestamp.

**Comment by hengfeiyang** [comment]
Yes, you are right, i add this to documents: https://docs.zincsearch.com/api/index/update-mapping/#custom-timestamp-field
